ONSdigital/design-system
# Search component fixes

### What is the context of this PR?
Fixes: #764

- Fixes issue with component not having the correct padding at screen width 500px when using the width constrained classes, removes some whitespace from the code and fixes some blank space output into the html around the classes on the input tag

- Fixes and issue with a red double error border when an error object is passed in, the input is focused and the character limit is reached

- Makes char check examples fit the amount of chars allowed to be entered

### How to review
- No blank space around classes on input and button
- Check search component looks correct at 500px screen width
- Check search component looks correct at all widths while adding `input--w-<number>` class. Testing numbers above and below 20
- Check other issues on card are fixed
- Pass an error object into the char check example and enter some test to go passed the char limit, then focus the input to make sure it behaves as other inputs when they have an error and has a black and yellow border
